packages/auto-label/src/auto-label.ts

```diff
@@ -148,6 +150,38 @@ handler.checkIfElementIsInArray = function checkIfElementIsInArray(
   }
 };
 
+// autoDetectLabel tries to detect the right api: label based on the issue
+// title.
+//
+// For example, an issue titled `spanner/transactions: TestSample failed` would
+// be labeled `api: spanner`.
+handler.autoDetectLabel = async (
+  github: GitHubAPI,
+  jsonArray: JSONData[],
+  issue: Webhooks.WebhookPayloadIssuesIssue,
+  owner: string,
+  repo: string
+): Promise<string | undefined> => {
+  if (!jsonArray) {
+    return undefined;
+  }
+  let firstPart = issue.title.split(':')[0]; // Before the colon, if there is one.
```

Comment:
Not sure we need to be *this* prescriptive about parsing titles.  Would just checking for an includes be terrible?

Reply:
Let's start here. I'd like to take a look at the logs after this and see what title formats don't parse, then we can update this logic.